src/community/jms-cluster/jms-commons/src/main/java/org/geoserver/cluster/client/JMSQueueListener.java

```diff
@@ -118,53 +105,48 @@ public void onMessage(Message message, Session session) throws JMSException {
             throw new IllegalArgumentException(
                     "Unable to handle a message without a generator class name");
         }
-        if (LOGGER.isLoggable(Level.FINE)) {
-            LOGGER.fine(
-                    "Incoming message was serialized using an handler generated by: \'"
-                            + generatorClass
-                            + "\'");
+        fine("Incoming message was serialized using an handler generated by: '%s'", generatorClass);
+
+        if (!(message instanceof ObjectMessage)) {
+            throw new JMSException("Unrecognized message type for catalog incoming event");
         }
 
         // USING INCOMING MESSAGE
-        if (message instanceof ObjectMessage) {
-
-            final ObjectMessage objMessage = (ObjectMessage) (message);
-            final Serializable obj = objMessage.getObject();
-
-            try {
-                // lookup the SPI handler, search is performed using the
-                // name
-                final JMSEventHandler<Serializable, Object> handler =
-                        jmsManager.getHandlerByClassName(generatorClass);
-                if (handler == null) {
-                    throw new JMSException(
-                            "Unable to find SPI named \'"
-                                    + generatorClass
-                                    + "\', be shure to load that SPI into your context.");
-                }
-
-                final Enumeration<String> keys = message.getPropertyNames();
-                final Properties options = new Properties();
-                while (keys.hasMoreElements()) {
-                    String key = keys.nextElement();
-                    options.put(key, message.getObjectProperty(key));
-                }
-                handler.setProperties(options);
-
-                // try to synchronize object locally
-                if (!handler.synchronize(handler.deserialize(obj))) {
-                    throw new JMSException(
-                            "Unable to synchronize message locally.\n SPI: " + generatorClass);
-                }
-
-            } catch (Exception e) {
-                final JMSException jmsE = new JMSException(e.getLocalizedMessage());
-                jmsE.initCause(e);
-                throw jmsE;
-            } finally {
-                this.consumedEvents.incrementAndGet();
+        final ObjectMessage objMessage = (ObjectMessage) (message);
```

Comment:
So basically here we always assume that indeed we get an ``ObjectMessage``, wondering any specific reason to get ride of the origin ``if``?

Reply:
We do, the factory used to create them creates only `ObjectMessage`s

Reply:
In fact, `JMSQueueListener` should implement `SessionAwareMessageListener<ObjectMessage>` not `SessionAwareMessageListener<Message>`